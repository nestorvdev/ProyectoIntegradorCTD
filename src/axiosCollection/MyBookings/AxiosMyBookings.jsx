import axios from "axios";
const baseURL = "http://worldguestbooking.com.ar:8080/"

function AxiosGetReservationsByUserId(setData, setLoading, setErrorMessage){
    axios.get(`${baseURL}reservations/get/user/${sessionStorage.getItem("id")}`)
      .then(response => {
        setData(response.data);
        setLoading(false);
        console.log(response.data, "data mybookings");
      })
      .catch(error => {
        setErrorMessage(error.message);
        setLoading(false);
      })
}

function AxiosDeleteReservation(reservationId, id, openModalSucceed){
    axios.delete(`${baseURL}reservations/delete/${reservationId}`)
            .then(response => {
                console.log(response.data, `delete reservation ${id}`);
                openModalSucceed()
                //window.location.href = "/mybookings";
            })
            .catch(error => {
                console.log(error.message);
            })
}

export { AxiosGetReservationsByUserId, AxiosDeleteReservation }