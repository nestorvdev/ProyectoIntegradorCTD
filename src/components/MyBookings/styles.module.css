a:visited,
a:link,
a:active {
  text-decoration: none;
}

.cardsBlock h2 {
  color: var(--secondary-color);
  font-size: var(--heading2);
  font-weight: var(--font-bold);
  margin-bottom: 9px;
  margin-top: 16px;
}

.cardsBlock {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 0 32px;
  padding-left: clamp(10px, 3.9%, 40px);
  padding-right: clamp(10px, 3.9%, 40px);
  min-height: 72vh;
}

.cardsBox {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.sinReservas{
  font-size: 22px;
  font-weight: 700;
  padding-top: 10px;
}

.cardBox {
  align-items: center;
  background-color: white;
  border: 1px solid #dfe4ea;
  border-radius: 10px;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  color: var(--secondary-color);
  display: flex;
  flex-wrap: wrap;
  height: 220px;
  justify-content: flex-start;
  margin: 5px 0;
  margin-right: 1px;
  width: clamp(590px, 50%, 630px);
  transition: all 0.3s;
}

.cardImage {
  background-size: cover;
  border-radius: 10px;
  height: 100%;
  position: relative;
  width: 47%;
}

.cardImage .image {
  border-radius: 10px;
  height: 100%;
  width: 100%;
  object-fit: cover;
  position: relative;
  z-index: 1;
}

.cardImage .iconHeart {
  position: absolute;
  right: 10px;
  top: 10px;
  transition: all 0.3s;
  width: 26px;
}

.cardImage {
  z-index: 90;
}

.iconHeart {
  z-index: 99;
}

/* cambiar color del corazon */
.heartColor {
  fill: rgba(0, 0, 0, 0.2);
  stroke: rgba(0, 0, 0, 0.5);
  stroke-width: 3px;
  filter: drop-shadow(4px 4px 1.5px rgba(0, 0, 0, 0.7));  
  stroke-dasharray: 75;
  stroke-dashoffset: 70;
  animation: dash 3s linear forwards;  
}

@keyframes dash {
  to {
    stroke-dashoffset: 0;
    stroke:white;
    fill: rgba(0, 0, 0, 0.05);
  }
}

.heartColor2 {
  fill: red;
  filter: drop-shadow(5px 3px 3px rgba(0, 0, 0, 0.7));
  stroke: red;
  
}

.cardImage .iconHeart:hover {
  transform: scale(1.1);
}

.cardImage .iconHeart:active {
  transform: scale(0.8);
}

/*ESTILOS MODAL FAVOURITE*/

.modalFavourite{
  display: flex;
  flex-direction: column;
  justify-content: center;
  height:22vh;
  width:39vw;
  border: 4px solid var(--primary-color); 

}

.modalFavourite p{
  color: black;
  font-size: 24px;
  font-weight: 900;
  text-align: center;
  padding: 10px 5px;  
}

.modalFavourite p a{
  text-decoration: black;
}

div.buttons {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  padding-right: 8px;
  height: 10vh;
}

div.buttons button, .clearFilterButton {
  background-color: white;
  border-radius: 5px;
  border: 2px solid  var(--primary-color);
  background-color: var(--primary-color);
  color: white;
  font-size: var(--button1);
  font-weight: var(--font-bold);
  padding: 5pt;
  width: clamp(164px,22.2vw,206px);
  transition: 0.5s;
}

div.buttons button.create{
  background-color: white;
  border: 2px solid  white;
  color: var(--primary-color);
}
div.buttons button.login:hover {
  background-color: #cf411d;
  border: 2px solid  #b9411f;
  cursor: pointer;

}
  
div.buttons button.create:hover {
  background-color: #ffd6c6;
  border: 2px solid  #ffd6c6;
  cursor: pointer;
}


.cardInfo {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  height: 93%;
  justify-content: space-between;
  margin-left: 4%;
  width: 47%;
}

.cardCategory {
  display: flex;
  font-size: var(--text3);
  text-transform: uppercase;
  width: 100%;
  justify-content: flex-start;
}

.cardCategory> div {
min-width: 80px;
}

.cardCategory .star {
  width: 14px;
  height: 14px;
}

.cardCategory p,
.iconLocation {
  margin-right: 4px;
  width: min-content;
}

.cardName {
  font-size: var(--heading3);
  font-weight: var(--font-bold);
  max-width: 100%;
}

.cardHeaderBox {
  align-items: flex-start;
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.cardHeadline,
.cardScore {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.cardHeadline {
  max-width: 75%;
}

.cardScore {
  align-items: flex-end;
  position: relative;
  z-index: 99;
}

.cardScoreWords {
  font-size: var(--text3);
  font-weight: var(--font-bold);
  overflow: hidden;
  white-space: nowrap;
}

.cardScoreNumber {
  align-items: center;
  background: var(--secondary-color);
  border-radius: 15px;
  color: white;
  display: flex;
  font-size: var(--heading3);
  font-weight: var(--font-bold);
  height: 35px;
  justify-content: center;
  min-width: 45px;
  padding: 8px 14px;
  text-align: center;
}

.cardLocation {
  font-size: var(--text3);
  font-weight: var(--font-bold);
}

.cardLocation span {
  color: var(--primary-color);
  font-weight: var(--font-bold);
  margin-left: 3px;
  text-transform: uppercase;
}

.cardLocation span:hover {
  cursor: pointer;
}

.cardIcons {
  display: flex;
  flex-wrap: wrap;
}

.cardFeatures {
  margin-right: 7px;
}

.cardDescription {
  font-size: var(--text3);
  font-weight: var(--font-bold);
  overflow: hidden;
  padding-top: 5px;
  display: flex;
  flex-direction: column;
}

.cardDescription p.noDesplegado {
  font-weight: var(--font-bold);
  max-height: 35px;
  overflow: hidden;
  padding: 0 2px; 

}

.cardDescription p.desplegado {
  max-height: 67px;
  overflow: hidden;
  padding: 0 2px;
}

.cardDescription span {
  color: var(--primary-color);
  font-weight: var(--font-bold);
  margin-left: 3px;
  
}

.cardDescription span:hover {
  cursor: pointer;
}
.link {
  width: 100%;
}

.cardButton2 {
  background-color: var(--primary-color);
  border: none;
  border-radius: 5px;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.12);
  color: white;
  font-size: var(--text3);
  font-weight: var(--font-bold);
  height: 50px;
  justify-content: center;
  text-align: center;
  transition: all 0.3s;
  width: 98%;
}


.clearFilterButton:hover {
  transform: scale(1.1);
  cursor: pointer;
}

.clearFilterButton:active {
  transform: scale(0.85);
}

.cardButton2:hover {
  transform: scale(1.02);
  cursor: pointer;
}

.cardButton2:active {
  transform: scale(0.85);
}

.buttonsBox {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.buttonsAdminBox {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
}

.cardButtonModify {
  background-color: var(--secondary-color);
  height: 24px;  
}

.pages {
  display: flex;
  justify-content: flex-end;
  font-size: var(--heading3);
  font-weight: var(--font-bold);
  padding: 5px;
}

.pages .left {
  width: 30px;
  transform: rotate(180deg);
  padding: 0 5px;
}

.pages .right {
  width: 30px;
  padding: 0 5px;
}

.pages button {
  padding: 0 10px;
  border: solid 1px var(--primary-color);
  margin: 0 5px;
  color: var(--primary-color);
  font-size: var(--text1);
  font-weight: var(--font-bold);
}

.pages button:hover {
  background-color: var(--primary-color);
  color: white;
  cursor: pointer;
}

.pages img:hover{
  background-color: white;
  color: white;
  cursor: pointer; 
}
.pages img:hover , .pages button:hover {
  pointer-events: painted;  
}

.pages button:disabled {
  pointer-events: none;
  background-color: var(--primary-color);
  color: white;
  border: none;
}


/*Estilos para el modalMap*/

.containerModalMap {
  width: 40vw;
}

/*Estilos para el botÃ³n de limpiar filtros */

.cardsBlock>h2 {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

.reservation{
  background-color: white;
}

@media all and (max-width:1350px){
  .cardsBox{
    justify-content: center;
  }
  
}

@media all and (max-width: 769px) {
  .cardsBlock h2 {
    margin-left: 1.8%;    
  }

  .cardsBlock {   
    min-height: 78vh;
  }

  .cardBox {
    max-width: 710px;
    min-width: 585px;
    width: 97vw;
  }

  .cardBox:nth-child(even) {
    margin-left: 0%;
  }

  .cardImage {
    height: 100%;
  }

  .modalFavourite{
    height:28vh;
    width: 50vw;
    }
}

@media all and (max-width: 610px) {
  .cardsBlock h2 {
    margin-left: 0%;
  }
  .cardBox {
    border: 0;
    height: 430px;
    justify-content: center;
    margin: 5px 0;
    min-width: 90vw;
    width: 95vw;
  }

  .cardImage {
    width: 100%;
    height: 47%;
  }
  .cardInfo {
    height: 47%;
    margin-left: 0%;
    margin-top: 3%;
    margin-bottom: 3%;
    width: 92%;
  }

  .cardIcons {
    display: none;    
  }

  .cardImage,
  .cardImage .image {
    border-radius: 0 0 10px 10px;
  }

  .cardImage .image {
    top: -2px;
    z-index: 0;
  }

  .clearFilterContainer {
    margin-top: 10px;
  }

  .cardsBlock>h2 {
    align-items: flex-start;
    flex-direction: column;
    justify-content: center;
  }
}

@media all and (max-width: 415px){
.modalFavourite{
  width: 70vw;
  height:21vh
  
  }
  .modalFavourite p{
    font-size: 18px ;
    font-weight: 700;
  }

  div.buttons button {
    font-size: 12pt;
    font-weight: var(--font-bold);
    padding: 3pt;
    transition: 0.5s;
    width: clamp(135px,22.2vw,206px);
  }
  .cardsBlock {   
    min-height: 70vh;
  }

}
