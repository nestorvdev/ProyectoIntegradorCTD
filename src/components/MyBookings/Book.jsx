import React, { useEffect, useState } from 'react';
import Styles from './styles.module.css';
import iconLocation from "./img/IconLocation.svg";
import { Link } from "react-router-dom";
import ScoreStar from '../Product/ScoreStar';
import ScoreDescription from '../Product/ScoreDescription';
import { Modal } from 'react-responsive-modal';
import ConfirmProductModal from '../Administrator/Product/ConfirmProductModal';
import ModalProductAviso from "../Administrator/Product/ModalProductSucceed";
import tildeOk from "../Administrator/icons/tildeOk.svg";
import { AxiosDeleteReservation } from '../../axiosCollection/MyBookings/AxiosMyBookings';
import axios from 'axios';

export default function Book({ id, startDate, endDate, reservationId }) {
    const role = sessionStorage.getItem("role");
    const [data, setData] = useState({});
    const [admin, setAdmin] = useState(true);
    const [loading, setLoading] = useState(true);
    const [modalConfirmDeletedIsOpen, setModalConfirmDeletedIsOpen] = useState(false)
    const [modalProductSucceedIsOpen, setModalProductSucceedIsOpen] = useState(false)

    useEffect(() => {
        axios.get(`http://worldguestbooking.com.ar:8080/products/get/${id}`)
            .then(response => {
                setData(() => response.data);
                console.log(response.data, "data Book");
                return response.data;
            })
            .then(() => setLoading(false))
    }, [])

    useEffect(() => {
        if (sessionStorage.getItem("role") === "ADMIN") {
            setAdmin(true)
        } else {
            setAdmin(false)
        }
    }, [role])

    const deleteReservation = (id) => {
        closeModalConfirmDeleted()
        AxiosDeleteReservation(reservationId, id,openModalSucceed)        
    }

    let loggued = sessionStorage.getItem("log");

    const openModalSucceed = (() => {
        setModalProductSucceedIsOpen(true)
    })

    const closeModalSucceed = (() => {
        setModalProductSucceedIsOpen(false)
        window.location.href = "/"
    })

    function openModalConfirmDeleted() {
        setModalConfirmDeletedIsOpen(true)
    }

    function closeModalConfirmDeleted() {
        setModalConfirmDeletedIsOpen(false)
    }

    return (
        <>{!loading &&
            <div className={Styles.cardBox} >
                <div className={Styles.cardImage}>
                    <img className={Styles.image} src={data.images.length > 0 ? data.images[0].url : ""} alt="" />
                </div>
                <div className={Styles.cardInfo}>
                    <div className={Styles.cardHeaderBox}>
                        <div className={Styles.cardHeadline}>
                            <div className={Styles.cardCategory}>
                                <p>{data.category.title}</p>
                                <ScoreStar qualification={data.qualification * 2} starColor="#F0572D" />
                            </div>
                            <div className={Styles.cardName}>{data.name}</div>
                        </div>
                        <div className={Styles.cardScore}>
                            <div className={Styles.cardScoreNumber}>
                                <p>{Math.floor(data.qualification * 2)}</p>
                            </div>
                            <div className={Styles.cardScoreWords}>
                                <ScoreDescription qualification={data.qualification * 2} />
                            </div>
                        </div>
                    </div>
                    <div className={Styles.cardLocation}>
                        <img className={Styles.iconLocation} src={iconLocation} alt="" />
                        {data.city.name},&#160;{data.city.country},&#160;{data.reference}
                    </div>

                    <div className={Styles.cardDescription}>
                        <p>Inicio de Reserva:{startDate} </p>
                        <p>Fin de Reserva:{endDate}</p>
                    </div>


                    <div className={Styles.buttonsBox}>
                        <Link to={`/product/${id}`} key={id} className={Styles.link}>
                            <button className={Styles.cardButton2}>Ver Locacion</button>
                        </Link>
                        <button className={Styles.cardButton2} onClick={openModalConfirmDeleted}>Eliminar</button>
                    </div>
                    <Modal open={modalConfirmDeletedIsOpen} onClose={closeModalConfirmDeleted} center>
                        <ConfirmProductModal accion="eliminar la reserva" setModalConfirmIsOpen={setModalConfirmDeletedIsOpen} funcionProducto={deleteReservation} closeModalConfirm={closeModalConfirmDeleted} />
                    </Modal>
                    <Modal open={modalProductSucceedIsOpen} onClose={closeModalSucceed} center>
                        <ModalProductAviso title="OperaciÃ³n confirmada." message="Se ha borrado la reserva exitosamente." closeModal={closeModalSucceed} icon={tildeOk} />
                    </Modal>
                </div>

            </div>
        }</>
    );
}