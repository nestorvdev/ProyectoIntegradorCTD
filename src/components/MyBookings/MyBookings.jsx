import React, { useState, useEffect } from "react";
import Styles from "./styles.module.css";
import StylesApp from "../../App.module.css";
import Book from "./Book.jsx";
import arrow from "./img/arrow.svg";
import TitleBar from "../Product/TitleBar";
import { AxiosGetProductosFavoritos } from "../../axiosCollection/Cards/AxiosCards";
import { AxiosGetReservationsByUserId } from "../../axiosCollection/MyBookings/AxiosMyBookings";


export default function MyBookings(props) {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);
  const [errorMessage, setErrorMessage] = useState("");
  const limitCardPerPage = 8;
  const [numberPage, setNumberPage] = useState(1);

  const [listadoFavoritos, setListadoFavoritos] = useState([]);

  useEffect(() => {
    AxiosGetProductosFavoritos(setListadoFavoritos, setErrorMessage);
    AxiosGetReservationsByUserId(setData, setLoading, setErrorMessage);    
    setNumberPage(1);
  }, []);


  const dataLimited = () =>
    data.slice(
      (numberPage - 1) * limitCardPerPage,
      numberPage * limitCardPerPage
    );

  const indexPages = () => {
    let pages = [];
    let cant =
      data.length % limitCardPerPage === 0
        ? data.length / limitCardPerPage
        : Math.floor(data.length / limitCardPerPage) + 1;
    for (let i = 0; i < cant; i++) {
      pages.push(
        <button
          onClick={() => setNumberPage(i + 1)}
          disabled={numberPage - 1 === i}
          key={i + 1}
        >
          {i + 1}
        </button>
      );
    }
    return pages;
  };


  return errorMessage && loading ? (
    <div className={`${StylesApp.delimiter}`}>
      <div className={`${Styles.cardsBlock} ${StylesApp.delimiterChild}`}>
        <h2>Reservas</h2>
        Resultados no disponibles - Falta la conexi√≥n con el Back
      </div>
    </div>
  ) : (
    <>
      <TitleBar name="Mis reservas" goBack={props.goBack} />
      <div className={`${StylesApp.delimiter}`}>
        <div className={`${Styles.cardsBlock} ${StylesApp.delimiterChild}`}>
          {data.length != 0 ?
            <div className={Styles.cardsBox}>{
             dataLimited().map((e) => (
              <Book
                key={e.id}
                id={e.productId}
                reservationId={e.id}
                startDate={e.startDate}
                endDate={e.endDate}
              />
              ))}
            </div>
            : <div className={Styles.sinReservas}>No hay reservas disponibles</div>}
          <div className={Styles.pages}>
            {numberPage > 1 && (
              <img
                className={Styles.left}
                onClick={() => setNumberPage(numberPage - 1)}
                src={arrow}
                alt="arrowLeft"
              />
            )}
            {indexPages()}
            {numberPage < indexPages().length && (
              <img
                className={Styles.right}
                onClick={() => setNumberPage(numberPage + 1)}
                src={arrow}
                alt="arrowRight"
              />
            )}
          </div>
        </div>
      </div>
    </>
  );
}
