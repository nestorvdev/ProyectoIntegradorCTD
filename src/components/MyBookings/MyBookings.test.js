import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import MyBookings from "./MyBookings";
import Book from "./Book";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <MyBookings/>", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<MyBookings/>)
    });

    it("Deberia mostrar <MyBookings/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    
})

describe("Probando el componente <Book/>", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Book/>)
    });

    it("Deberia mostrar <Book/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
})