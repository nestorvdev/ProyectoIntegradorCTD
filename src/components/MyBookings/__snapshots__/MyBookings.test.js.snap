// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <Book/> Deberia mostrar <Book/> correctamente 1`] = `<Fragment />`;

exports[`Probando el componente <MyBookings/> Deberia mostrar <MyBookings/> correctamente 1`] = `
<Fragment>
  <TitleBar
    name="Mis reservas"
  />
  <div
    className="delimiter"
  >
    <div
      className="cardsBlock delimiterChild"
    >
      <div
        className="sinReservas"
      >
        No hay reservas disponibles
      </div>
      <div
        className="pages"
      />
    </div>
  </div>
</Fragment>
`;
